aws/resource_aws_redshift_cluster.go

```diff
@@ -711,17 +747,20 @@ func resourceAwsRedshiftClusterUpdate(d *schema.ResourceData, meta interface{})
 		}
 	}
 
-	if d.HasChange("enable_logging") || d.HasChange("bucket_name") || d.HasChange("s3_key_prefix") {
+	deprecatedHasChange := (d.HasChange("enable_logging") || d.HasChange("bucket_name") || d.HasChange("s3_key_prefix"))
+	if d.HasChange("logging") || deprecatedHasChange {
 		var loggingErr error
-		if _, ok := d.GetOk("enable_logging"); ok {
 
+		logging, ok := d.GetOk("logging.0.enable")
+		_, deprecatedOk := d.GetOk("enable_logging")
+
+		if (ok && logging.(bool)) || deprecatedOk {
```

Comment:
I think you want to check the value of `enable_logging`, rather than that it's set to a value?

Reply:
As it is a boolean field it shouldn't matter, b/c it will always be present - that is in fact why we introduced [`d.GetOkExists`](https://github.com/hashicorp/terraform/blob/4cde21501c50dab41425569da7bfc71861289937/helper/schema/resource_data.go#L109), secondly I intentionally left the implementation as is since it will be deprecated eventually.